integrations/api_repo_test.go

```diff
@@ -33,22 +32,19 @@ func TestAPIUserReposNotLogin(t *testing.T) {
 	}
 }
 
-type searchResponseBody struct {
-	ok   bool
-	data []api.Repository
-}
-
 func TestAPISearchRepoNotLogin(t *testing.T) {
 	prepareTestEnv(t)
 	const keyword = "test"
 
 	req := NewRequestf(t, "GET", "/api/v1/repos/search?q=%s", keyword)
 	resp := MakeRequest(t, req, http.StatusOK)
 
-	var body searchResponseBody
+	var body api.SearchResults
 	DecodeJSON(t, resp, &body)
-	for _, repo := range body.data {
-		assert.True(t, strings.Contains(repo.Name, keyword))
+	assert.NotEmpty(t, body.Data)
```

Comment:
`require.NotEmpty` ?

Reply:
@bkcsoft I don't want to stop tests.

Reply:
@Morlinest Why? One failing test is enough to let whole integration tests fail. It will speed up testing a lot.

Reply:
@daviian If you are thinking about that like this, we should use `require.*` everywhere :). I want to see more than 1 step ahead. In my opinion, it's like parsing a form and showing just first error to user.

Reply:
@Morlinest in this particular test every following assertion will evaluate to false so there is no reason to check them anyway

Reply:
@daviian Following assertions are inside range loop.